helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PerInstanceAccessor.java

```diff
@@ -308,19 +310,33 @@ public Response updateInstanceConfig(@PathParam("clusterId") String clusterId,
     }
     InstanceConfig instanceConfig = new InstanceConfig(record);
     ConfigAccessor configAccessor = getConfigAccessor();
+
+    if (doSanityCheck && (command == Command.delete || command == Command.update)) {
+      try {
+        validateDeltaInstanceConfigForUpdate(clusterId, instanceName, configAccessor,
+            instanceConfig, command == Command.delete);
+      } catch (IllegalArgumentException ex) {
+        LOG.error(
+            String.format("Error in update instance config for instance: %s", instanceName),
+            ex);
+        return serverError(ex);
```

Comment:
If possible, let us return some useful error information to the caller, for example, where is the format error, such as "Invalid topology setting", etc.